# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/events/Touch.cpp
# Commit: 0ea945553f63
# Full Hash: 0ea945553f63659b99fd37265c5e7a764f70504d
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2024-12-03 16:21:08
# Regressor Bug: 1680669
# File Overlap Count: 1
# Description:
#   Bug 1680669 - part 6: Make methods of `MouseEvent` and static methods of `Event` aware of fractional coordinates r=smaug,emilio,webdriver-reviewers
#   
#   `screenX`, `screenY`, `clientX`, `clientY`, `x`, `y`, `offsetX` and `offsetY`
#   are now `double`.  This patch makes the event classes aware of DOM events.
#   
# ==============================================================================

diff -r 98d5a05c6c6e -r 0ea945553f63 dom/events/Touch.cpp
--- a/dom/events/Touch.cpp	Tue Dec 03 07:10:01 2024 +0000
+++ b/dom/events/Touch.cpp	Tue Dec 03 07:10:05 2024 +0000
@@ -187,12 +187,12 @@
   if (mPointsInitialized) {
     return;
   }
-  mClientPoint =
-      Event::GetClientCoords(aPresContext, aEvent, mRefPoint, mClientPoint);
-  mPagePoint =
-      Event::GetPageCoords(aPresContext, aEvent, mRefPoint, mClientPoint);
-  mScreenPoint =
-      Event::GetScreenCoords(aPresContext, aEvent, mRefPoint).extract();
+  mClientPoint = RoundedToInt(
+      Event::GetClientCoords(aPresContext, aEvent, mRefPoint, mClientPoint));
+  mPagePoint = RoundedToInt(
+      Event::GetPageCoords(aPresContext, aEvent, mRefPoint, mClientPoint));
+  mScreenPoint = RoundedToInt(
+      Event::GetScreenCoords(aPresContext, aEvent, mRefPoint).extract());
   mPointsInitialized = true;
 }
 